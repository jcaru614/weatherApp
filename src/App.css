body {
  background: url('./images/background2.png');
  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
}
