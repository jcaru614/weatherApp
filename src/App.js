import Home from './Components/Home/index';
import { Provider } from "react-redux";
import store from "./store"

function App() {
  return (
    <Provider store={store}>
      <Home />
    </Provider>

  );
}

export default App;
