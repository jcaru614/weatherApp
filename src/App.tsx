import React, { Component } from 'react';
import Home from './Components/Home/index';
import { Provider } from "react-redux";
import store from "./store"

class App extends Component {
  render() {
    return (
      <Provider store={store}>
          <Home />
      </Provider>
    );
  }

}

export default App;
