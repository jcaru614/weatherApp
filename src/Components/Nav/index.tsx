import React from 'react';
import { makeStyles, Theme } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import Typography from '@material-ui/core/Typography';
import Box from '@material-ui/core/Box';
import Current from '../Current/index';

interface TabPanelProps {
  children?: React.ReactNode;
  dir?: string;
  index: any;
  value: any;
}

function TabPanel(props: TabPanelProps) {
  const { children, value, index, ...other } = props;

  return (
    <div
      role="tabpanel"
      hidden={value !== index}
      id={`full-width-tabpanel-${index}`}
      aria-labelledby={`full-width-tab-${index}`}
      {...other}
    >
      {value === index && (
        <Box p={3}>
          <Typography>{children}</Typography>
        </Box>
      )}
    </div>
  );
}

function a11yProps(index: any) {
  return {
    id: `full-width-tab-${index}`,
    'aria-controls': `full-width-tabpanel-${index}`,
  };
}

const useStyles = makeStyles((theme: Theme) => ({
  root: {
    flexGrow: 1,
    width: 570,
    position: 'absolute',
    top: '10%',
    left: '35.5%',
    marginTop: '-50px',
    marginLeft: '-50px',
  },
}));

export default function FullWidthTabs() {
  const classes = useStyles();
  const [value, setValue] = React.useState(0);

  const handleChange = (event: React.ChangeEvent<{}>, newValue: number) => {
    setValue(newValue);
  };

  return (
    <>
      <div className={classes.root}>
        <AppBar position="static" color="primary">
          <Tabs
            value={value}
            onChange={handleChange}
            indicatorColor="secondary"
            textColor="secondary"
            variant="fullWidth"
          >
            <Tab label="Current Weather" {...a11yProps(0)} />
            <Tab label="Hourly Forecast" {...a11yProps(1)} />
            <Tab label="Road Risk" {...a11yProps(2)} />
          </Tabs>
        </AppBar>
      </div>
      <TabPanel value={value} index={0} >
        <Current />
        </TabPanel>
      <TabPanel value={value} index={1} >
        Item Two
        </TabPanel>
      <TabPanel value={value} index={2} >
        Item Three
        </TabPanel>


    </>
  );
}
