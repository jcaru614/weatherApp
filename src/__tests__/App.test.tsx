import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import App from '../App';
import Nav from '../Components/Nav/index';

// configure({adapter: new Adapter()});

let wrapped:any;
beforeEach(() => {
    wrapped = shallow(<App />);
})
it('renders the component', () => {
    expect(wrapped.find(Nav).length).toEqual(1)
  });

  it('matches snapshot', () => {
    let component = shallow(<App />);
    expect(component).toMatchSnapshot();
  });