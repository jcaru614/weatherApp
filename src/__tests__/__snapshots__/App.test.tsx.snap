// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <FullWidthTabs />
</Provider>
`;
